### ISSUE TITLE
How can I get IMG_MEAN of a custom dataset?

### ISSUE THREAD

**Issue by CaiziLee** [DrSleep/tensorflow-deeplab-resnet#146]
dear @DrSleep , if there is a custom dataset, how can i get the IMG_MEAN, thank you

**Comment by siinem** [comment]
@CaiziLee According to #106, it is computed through the training images.

**Comment by siinem** [comment]
@DrSleep I just wonder is the mean vector in the form of RGB or BGR ?

**Comment by CaiziLee** [comment]
@siinem compute the sum of each channel of the training images and then divided by the count ? but if it's necessary to recompute IMG_MEAN when processing the test samples

**Comment by siinem** [comment]
Hi @DrSleep,
When i run the fine_tune.py file for my dataset with mean value of the training images **in the format of RGB**, i got these performance results (I have 637 images in training set, and 352 images attest set, with 66 categories - one category is background- ):
on training set: Mean IoU: 0.579
on test set: Mean IoU: 0.212

But, when i run the fine_tune.py file for my dataset with mean value of the training images **in the format of BGR**, i got these performance results :
on training set: Mean IoU: 0.173
on test set: Mean IoU: 0.232

Btw. for both cases i also change the mean vector at the evaluate.py file.

So. do you think that maybe it is accepting mean vector in RGB format? I wondered your comments on these performance results.

**Comment by siinem** [comment]
@DrSleep @CaiziLee @arslan-chaudhry 
I have just computed the mean of pascal voc 2012 training images by matlab.
I found that the mean of the 10582 images given at the train.txt file is : **[116.6234  111.5128  103.1481] in RGB format** at matlab.

The exploited mean value at the published codes is as (it is expected to be computed for the pascal voc 2012 trainign images) :
IMG_MEAN = np.array((104.00698793,116.66876762,122.67891434), dtype=np.float32)

How i compute the mean values in matlab is as below:

```
path_imnames = '.\train.txt';
path_images = '.\VOCdevkit\VOC2012';

fid = fopen(path_imnames);
tline = fgetl(fid);

i=0;
while ischar(tline)
    i = i+1;
    im_name = split(tline," ");
    im_name = im_name{1};
    
    im = imread(fullfile(path_images,im_name));
    ch1 = im(:,:,1);
    ch2 = im(:,:,2);
    ch3 = im(:,:,3);
    MEAN_IMG(i,:) = mean([ch1(:), ch2(:), ch3(:)],1);
    tline = fgetl(fid);
end
fclose(fid);
MEAN_IMG_ = mean(MEAN_IMG,1)
```
So, i really wonder how the values of IMG_MEAN at the code are computed?

**Comment by siinem** [comment]
After searching internet, I found that IMG_MEAN = np.array((104.00698793,116.66876762,122.67891434), dtype=np.float32) is not computed for PASCAL VOC trainign images, but these are the mean values computed for the ImageNet Dataset in BGR format ([link](https://github.com/bwasti/AICamera/issues/13 ) ) . Thus, i guess that we SHOULD NOT change these values when tuning to our dataset, is it true??

**Comment by dimitriGallos** [comment]
I empirically found that it's better to let IMG_MEAN to be what it is in the code and not change it if you have a small dataset (mine is ~1000 images).

**Comment by DrSleep** [comment]
usually, segmentation networks are being pre-trained for image classification, hence a particular set of normalisation parameters from image classification tends to be used for segmentation as well. if you train from scratch, it is better to use statistics from your current training set.
